tests/sqlgateway/mock/mock_client.py

```diff
@@ -0,0 +1,62 @@
+import os
+from typing import List
+
+from flink.sqlgateway.operation import SqlGatewayOperation
+from flink.sqlgateway.session import SqlGatewaySession
+from flink.sqlgateway.config import SqlGatewayConfig
+from tests.sqlgateway.mock.gw_router import GwRouter
+import requests
+import json
+
+
+class MockFlinkSqlGatewayClient:
+    def __init__(self):
+        self.router = None
+        self.session = None
+
+    def create_session(self, host: str, port: int, session_name: str) -> SqlGatewaySession:
+        host_port = f"http://{host}:{port}"
+        test_config = {
+            "host_port": host_port,
+            "schemas": [
+                {"catalog": "default_catalog", "database": "default_database", "tables": [], "views": []}
+            ],
+            "current_catalog": "default_catalog",
+            "current_database": "default_database",
+        }
+        self.router = GwRouter(test_config)
+        self.router.start()
+        # create session
+        r = requests.post(f"{host_port}/v1/sessions", json.dumps({"sessionName": f"{session_name}"}))
```

Comment:
This seems like not actually used. It will be the real `flink.sqlgateway.client.FlinkSqlGatewayClient` that will be opening the session anyway.